api/v1beta1/azuremanagedcontrolplane_default_test.go

```diff
@@ -53,3 +54,90 @@ func hardcodedAzureManagedControlPlaneWithSSHKey(sshPublicKey string) *AzureMana
 		},
 	}
 }
+
+func TestSetDefaultAutoScalerProfile(t *testing.T) {
+	g := NewWithT(t)
+
+	type test struct {
+		amcp *AzureManagedControlPlane
+	}
+
+	defaultAMP := &AzureManagedControlPlane{
+		Spec: AzureManagedControlPlaneSpec{
+			AutoScalerProfile: &AutoScalerProfile{
+				BalanceSimilarNodeGroups:      (*BalanceSimilarNodeGroups)(pointer.String(string(BalanceSimilarNodeGroupsFalse))),
+				Expander:                      (*Expander)(pointer.String(string(ExpanderRandom))),
+				MaxEmptyBulkDelete:            pointer.String("10"),
+				MaxGracefulTerminationSec:     pointer.String("600"),
+				MaxNodeProvisionTime:          pointer.String("15m"),
+				MaxTotalUnreadyPercentage:     pointer.String("45"),
+				NewPodScaleUpDelay:            pointer.String("0s"),
+				OkTotalUnreadyCount:           pointer.String("3"),
+				ScanInterval:                  pointer.String("10s"),
+				ScaleDownDelayAfterAdd:        pointer.String("10m"),
+				ScaleDownDelayAfterDelete:     pointer.String("10s"),
+				ScaleDownDelayAfterFailure:    pointer.String("3m"),
+				ScaleDownUnneededTime:         pointer.String("10m"),
+				ScaleDownUnreadyTime:          pointer.String("20m"),
+				ScaleDownUtilizationThreshold: pointer.String("0.5"),
+				SkipNodesWithLocalStorage:     (*SkipNodesWithLocalStorage)(pointer.String(string(SkipNodesWithLocalStorageFalse))),
+				SkipNodesWithSystemPods:       (*SkipNodesWithSystemPods)(pointer.String(string(SkipNodesWithSystemPodsTrue))),
+			},
+		},
+	}
+
+	allFieldsAreNilTest := test{amcp: &AzureManagedControlPlane{
+		Spec: AzureManagedControlPlaneSpec{
+			AutoScalerProfile: &AutoScalerProfile{},
+		},
+	}}
+
+	allFieldsAreNilTest.amcp.setDefaultAutoScalerProfile()
+
+	g.Expect(allFieldsAreNilTest.amcp.Spec.AutoScalerProfile).To(Equal(defaultAMP.Spec.AutoScalerProfile))
+
+	allFieldsAreNotNilTest := test{amcp: &AzureManagedControlPlane{
+		Spec: AzureManagedControlPlaneSpec{
+			AutoScalerProfile: &AutoScalerProfile{
```

Comment:
I think it would make the code a bit cleaner to set the AutoScalerProfile to a variable, then compare that to the outcome to ensure none of the fields have changed. That would save a few lines of code and not require a comparison for each individual field.

Reply:
Done. Thanks for the suggestion.

Reply:
Since the `AutoScalerProfile` is equivalent to `expectedNotNil`, it should work out to set `AutoScalerProfile` to expectedNotNil, then ensuring it is still equal to `expectedNotNil` after calling `setDefaultAutoScalerProfile()`. That way, you don't have to repeat setting the long struct.

Reply:
I have made the changes